### DISCUSSION TITLE
Image alignment improvements

### DISCUSSION THREAD

**Discussion by binaryfire** [outline/outline#5112] [Ideas]
Hey Tom

Would it be possible to add some image alignment options that don't wrap? I often want to align an image left without any text wrapping on the right. Currently there's no clean way of doing this. 

Also, even when you align an image to the left and insert blank lines on the right, the image can't be enlarged.

**Comment by florian-obradovic** [comment]
Same for me.
![CleanShot 2023-11-16 at 19 41 41@2x](https://github.com/outline/outline/assets/57985125/0c4153d6-abc2-42c0-b445-cd2abe142f17)

Reported here: https://github.com/outline/outline/issues/3403

**Comment by florian-obradovic** [comment]
Is there any other info you need?

**Comment by threema-danilo** [comment]
Ideally there would be three aspects that could be controlled separately, similar to most text editing software:

- Size: Either allow specifying an arbitrary size, or offer options like "Small / Medium / Large / Full-Width".
- Alignment/Flow: Left aligned (no float), let aligned (float), center aligned (no float), right aligned (no float), right aligned (float)

(Note: The icon for center alignment currently shows text to the left and right of the image, even though it functions like a block element and there is no flow of text around the image. This could be considered a bug.)

**Reply by florian-obradovic** [reply]
This it how it works in Confluence onPrem (not cloud). It's not perfect but I like it very much. Especially that I can post a few smaller screenshots side by side.
I also like how it matches the size if I choose medium and paste another pic, choose medium again and they match up in height.

In outline currently it's impossible to have screenshots side by side, right?

![CleanShot 2024-02-19 at 21 00 41](https://github.com/outline/outline/assets/57985125/047c92f8-bde1-416a-abfa-b4d2b76b862c)

**Reply by threema-danilo** [reply]
Yup, I'm also coming from Confluence OnPrem, and while Confluence had a lot of quirks, the image handling was pretty nice and I miss it.

**Comment by ovizii** [comment]
This has been driving me insane, until I noticed it wasn't me but rather Outline at fault :-) 
A bit of googling and I found related issues: https://github.com/outline/outline/issues/3403

I also found this rather old discussion which claims resizing has been implemented since v0.67.0+ but I don't see it. Could someone please share a screenshot if you have the resizing option available anywhere? 
https://github.com/outline/outline/discussions/1369#discussioncomment-4663115

Forgot to add I'm self-hosting Outline in case it matters.

**Reply by fyksen** [reply]
[Screencast from 2024-03-02 21-08-31.webm](https://github.com/outline/outline/assets/4790633/6ee1a783-20e2-40a5-a071-4fb4367cf53e)

Resizing works.

**Reply by ovizii** [reply]
Damn, I totally did not realize that is what those bars on the left and right side of the image do.
So that leaves the initial posters request about the alignment problems.

**Comment by florian-obradovic** [comment]
Looks like it got somehow fixed in #6350 :)
Thanks - it would be killer to have a setting to have images always aligned left and put text below, not beside.

**Comment by florian-obradovic** [comment]
Picking this up again because it's still an issue for me:
https://github.com/outline/outline/discussions/11430
